pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2858,10 +2883,11 @@ public void endReplaceSegments(String tableNameWithType, String segmentLineageEn
             .format("Invalid segment lineage entry id (tableName='%s', segmentLineageEntryId='%s')", tableNameWithType,
                 segmentLineageEntryId));
 
-        // NO-OPS if the entry is already completed
-        if (lineageEntry.getState() == LineageEntryState.COMPLETED) {
-          LOGGER.warn("Lineage entry state is already COMPLETED. Nothing to update. (tableNameWithType={}, "
-              + "segmentLineageEntryId={})", tableNameWithType, segmentLineageEntryId);
+        // NO-OPS if the entry is already 'COMPLETED' or 'REVERTED'
+        if (lineageEntry.getState() != LineageEntryState.IN_PROGRESS) {
+          LOGGER.warn("Lineage entry state is not IN_PROGRESS. Nothing to update. (tableNameWithType={}, "
```

Comment:
I suggest rephrasing this sentence a bit, or add a message at the very beginning of the method in order to show that we are trying to mark the lineage entry to `COMPLETED`. Otherwise, it's a bit hard to track what happened when seeing this message.

Same for start and revert APIs.

Reply:
fixed